webpack/components/RoutedTabs/index.js

```diff
@@ -5,37 +5,44 @@ import { Switch, Route, Redirect, useHistory, useLocation, withRouter, HashRoute
 import { head, last } from 'lodash';
 
 const RoutedTabs = ({
-  tabs, defaultTabIndex, titleComponent,
+  tabs, defaultTabIndex,
 }) => {
   const { push } = useHistory();
   const {
     hash, key: locationKey,
   } = useLocation();
 
-  // The below transforms #/filters/6 to filters
+  // The below transforms #/history/6 to history
   const currentTabFromUrl = head(last(hash.split('#/')).split('/'));
 
-  const onSelect = (_e, eventKey) => {
-    // This prevents needless pushing on repeated clicks of a tab
-    if (hash.slice(2) !== eventKey) {
-      push(`#/${eventKey}`);
+  const onSelect = (e, key) => {
+    e.preventDefault();
+    // Check if the middle mouse button is used, if so do nothing.
+    // Check if the current hash contains the key, if so do nothing.
+    if (!(e.button === 1 || e.buttons === 4) && hash.slice(2) !== key) {
+      push(`#/${key}`);
     }
   };
 
+
   return (
     <>
       <Tabs
         activeKey={currentTabFromUrl}
-        onSelect={onSelect}
       >
-        {tabs.map(({ key, title }) =>
-        (
-          <Tab
-            aria-label={title}
-            eventKey={key}
+        {tabs.map(({ key, title }) => (
+          <a
             key={key}
-            title={titleComponent || <TabTitleText>{title}</TabTitleText>}
-          />
+            href={`#/${key}`}
+            onMouseDown={e => onSelect(e, key)}
```

Comment:
Should this be `onMouseUp`? On other web pages this seems to be triggered for me on mouse up, not down.

Reply:
Sure!